### PR TITLE
fix: slow client recovery in tests

### PR DESCRIPTION
**Pull request by dpc**
I've noticed recovery tests are still super slow and investigating why that is, found bunch of bugs. This was the first time we are actually using api version discovery for anything, so it's unsurpring that I've found bunch of (my own) bugs. :D

### CODE REVIEW

File: fedimint-client/src/module/init.rs [hunk 1]
```diff
@@ -57,8 +58,17 @@ where
...
+    }
+
+    pub fn module_api_version(&self) -> &ApiVersion {
+        &self.module_api_version
+    }
+
+    // TODO: deprecate, use `module_api_version` instead
     pub fn api_version(&self) -> &ApiVersion {
-        &self.api_version
+        &self.module_api_version
```
Comment (elsirion) [inline fedimint-client/src/module/init.rs:71]: Why not just remove? Because backporting?

**Review by elsirion — APPROVED** [review]

File: fedimint-client/src/module/init.rs (hunk 1 above)
Comment (dpc) [reply fedimint-client/src/module/init.rs:71]: Just wanted to make the change small. But now thinking about it - probably will help with backports indeed.

**Review by bradleystachurski — APPROVED** [review]

File: devimint/src/tests.rs [hunk 2]
```diff
@@ -312,7 +312,7 @@ pub async fn latency_tests(dev_fed: DevFed, r#type: LatencyTest) -> Result<()> {
             let restore_time = start_time.elapsed();
 
             println!("### LATENCY RESTORE: {restore_time:?}");
-            assert!(restore_time < Duration::from_secs(160));
+            assert!(restore_time < Duration::from_secs(10));
```
[the remark this answers is not in the document]
Comment (justinmoon) [reply devimint/src/tests.rs:315]: 🏩

[the remark this answers is not in the document]
Comment (elsirion) [reply devimint/src/tests.rs:315]: @justinmoon TIL about this emoji :rofl:

**Pull request merged**